### PR TITLE
Include resource labels in Zipkin export

### PR DESCRIPTION
**Pull request by pmm-sumo**
**Description:** 

Resource data might include various labels that are currently missed when doing conversion. It seems to be a current limitation of the design: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/exporter-zipkin.md

This change adds this capability, simple merging resource labels with span attributes

**Link to tracking Issue:** N/A

**Testing:** Unit Tests + Manual tests on interoperability with K8S metadata tagging

**Documentation:** Relevant project documentation updated

### CODE REVIEW

File: exporter/zipkinexporter/config.go [hunk 1]
```diff
@@ -27,5 +27,8 @@ type Config struct {
 	URL    string `mapstructure:"url"`
 	Format string `mapstructure:"format"`
 
+	// Whether resource labels from TraceData are to be included in Span
+	ExportResourceLabels string `mapstructure:"export_resource_labels"`
```
Comment (pjanotti) [inline exporter/zipkinexporter/config.go]: Why not make this a `bool`?

File: translator/trace/spandata/protospan_to_spandata.go [hunk 2]
```diff
@@ -129,12 +131,19 @@ func protoLinkTypeToOCLinkType(lt tracepb.Span_Link_Type) trace.LinkType {
 	}
 }
 
-func protoAttributesToOCAttributes(attrs *tracepb.Span_Attributes) map[string]interface{} {
+func protoAttributesToOCAttributes(attrs *tracepb.Span_Attributes, resource *resourcepb.Resource) map[string]interface{} {
 	if attrs == nil {
```
Comment (pjanotti) [inline translator/trace/spandata/protospan_to_spandata.go]: if there are no other span attributes but there are resources they will be missed - granted it is a corner case but possible.

File: exporter/zipkinexporter/config.go (hunk 1 above)
Comment (pmm-sumo) [reply exporter/zipkinexporter/config.go]: Good point, I thought that the case with not-set value would not be handled but `mapstructure` is smart enough to handle this when `*bool` is used

**Comment by pmm-sumo** [comment]
Many thanks for the review @pjanotti, it was very helpful. Still new to Go world, the comments were very helpful 

> LGTM @pmm-sumo one last thing: could you please add the new config setting to
> 
> https://github.com/open-telemetry/opentelemetry-collector/blob/6c299e1bc8221410125755abc18322971e292e8d/exporter/zipkinexporter/testdata/config.yaml#L10
> 
> set to false and update the respective test that is going to break when you do that.

I actually set it to false before and now just added a basic assertion: https://github.com/open-telemetry/opentelemetry-collector/pull/591/files#diff-a517b938bf41e95d815cb5104ae56719R51

Not sure why it was supposed to break (just adding the flag did not make it fail) - perhaps I am missing something?

File: exporter/zipkinexporter/zipkin.go [hunk 3]
```diff
@@ -104,8 +113,14 @@ func createZipkinExporter(logger *zap.Logger, config configmodels.Exporter) (*zi
 
 func (ze *zipkinExporter) PushTraceData(ctx context.Context, td consumerdata.TraceData) (droppedSpans int, err error) {
 	tbatch := []*zipkinmodel.SpanModel{}
+
+	var resource *resourcepb.Resource = nil
```
Comment (pjanotti) [inline exporter/zipkinexporter/zipkin.go]: nit: in go typically we don't assign the zero value, so 
```suggestion
	var resource *resourcepb.Resource
```

**Comment by pjanotti** [comment]
> Not sure why it was supposed to break (just adding the flag did not make it fail) - perhaps I am missing something?

I expected it to break because the zero value is `nil` and typically those tests compare the whole data structure, but that is not the case for this exporter in particular: the test checks each field explicitly. Adding the assertion is good enough in this case.

**Comment by pjanotti** [comment]
@pmm-sumo could you please sign the CLA?

**Comment by pmm-sumo** [comment]
> @pmm-sumo could you please sign the CLA?

Of course, I am waiting for my organisation to handle signing up us a contributor, hope to have that accomplished soon

**Comment by pmm-sumo** [comment]
I signed it

File: exporter/zipkinexporter/config.go [hunk 4]
```diff
@@ -27,5 +27,8 @@ type Config struct {
 	URL    string `mapstructure:"url"`
 	Format string `mapstructure:"format"`
 
+	// Whether resource labels from TraceData are to be included in Span. True by default
+	ExportResourceLabels *bool `mapstructure:"export_resource_labels"`
```
[the remark this answers is not in the document]
Comment (pmm-sumo) [reply exporter/zipkinexporter/config.go]: I created the issue: https://github.com/open-telemetry/opentelemetry-collector/issues/595 and referenced it in the comments. Will be glad to remove it (say, ~April 5th?)

**Pull request merged**